Jenkinsfile-deploy

```diff
@@ -0,0 +1,59 @@
+pipeline {
+  agent none
+  parameters {
+    choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are building the front end for")
+  }
+  stages {
+    stage("Test") {
+        agent {
+            ecs {
+                inheritFrom "sbt"
+            }
+        }
+        steps {
+            checkout scm
+            sh "sbt dist"
+            stash includes: "Dockerfile", name: "Dockerfile"
+            stash includes: "target/universal/tdr-transfer-frontend-*.zip", name: "tdr-transfer-frontend-zip"
+        }
+    }
+    stage("Docker") {
+        agent {
+            label "master"
+        }
+        steps {
+            unstash "tdr-transfer-frontend-zip"
+            unstash "Dockerfile"
+            sh "docker build -t nationalarchives/tdr-transfer-frontend:${params.STAGE} ."
+            withCredentials([usernamePassword(credentialsId: "docker", usernameVariable: "USERNAME", passwordVariable: "PASSWORD")]) {
```

Comment:
Where do the `USERNAME` and `PASSWORD` environment variables come from, and should we be using something like the [docker-workflow](https://jenkins.io/doc/pipeline/steps/docker-workflow/) plugin to avoid referencing the credentials directly?

Reply:
USERNAME and PASSWORD are created temporarily by the withCredentials plugin. They only exist for this job inside that block. 
We could use something like that but it's a lot of overhead just to avoid printing out obfuscated credentials.

Reply:
It's actually not a lot of effort to use the plugin and it's cleaner so I've added it back in.

Reply:
Great! 👍